react/NestedSelect/NestedSelect.jsx

```diff
@@ -0,0 +1,162 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import CompositeRow from '../CompositeRow'
+import Icon from '../Icon'
+import styles from './styles.styl'
+import UIRadio from '../Radio'
+import cx from 'classnames'
+import omit from 'lodash/omit'
+
+const Radio = ({ className, ...props }) => (
+  <UIRadio className={cx(styles.Radio, className)} {...props} />
+)
+
+const Divider = () => <div className={styles.Divider} />
+
+const ItemRow = ({ item, onClick, isSelected }) => {
+  return (
+    <div className={cx(styles.Row, isSelected ? styles.Row__selected : null)}>
+      <CompositeRow
+        dense
+        image={item.icon}
+        primaryText={item.title}
+        onClick={() => onClick(item)}
+        right={
+          item.children && item.children.length > 0 ? (
+            <Icon icon="right" color="var(--coolGrey)" />
+          ) : (
+            <Radio checked={isSelected} />
+          )
+        }
+      />
+    </div>
+  )
+}
+
+/**
+ * Select like component to choose an option among a list of options.
+ * Options can have children; selecting an option that has children
+ * will show the children of the chosen option instead of selecting
+ * the option.
+ */
+class NestedSelect extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      history: [props.options]
+    }
+  }
+
+  componentWillUnmount() {
+    this.unmounted = true
+  }
+
+  resetHistory() {
+    if (this.unmounted) {
+      return
+    }
+    this.setState({ history: [this.props.options] })
+  }
+
+  handleBack = () => {
+    const [item, ...newHistory] = this.state.history
+    this.setState({
+      history: newHistory
+    })
+    return item
+  }
+
+  handleNavToChildren = item => {
+    const newHistory = [item, ...this.state.history]
+    this.setState({
+      history: newHistory
+    })
+  }
+
+  handleSelect = item => {
+    this.props.onSelect(item)
+    setTimeout(() => {
+      this.resetHistory()
+    }, 500)
```

Comment:
Why do we need this?

Reply:
Thanks for asking, it is actually a bit weird. I commented in https://github.com/cozy/cozy-ui/pull/1270/commits/aceb1d0f4130056a8da9325a84e2ccd500788021. I am sure we will be able to fix this when using a better transition system.